bin/install-pr

```diff
@@ -9,63 +9,193 @@
+
+# Read script flags and arguments.
+while :
+do
+  case $1 in
+    -h|--help)
+      echo "Install Linkerd with the changes made in a GitHub Pull Request."
+      echo ""
+      echo "Note:"
+      echo "    If installing into an existing Kubernetes cluster, this script"
+      echo "    assumes the cluster is not a KinD cluster."
+      echo ""
+      echo "Usage:"
+      echo "    # Install Linkerd into the 'kubectl' current cluster"
+      echo "    bin/install-pr ####"
+      echo ""
+      echo "    # Install Linkerd into a new KinD cluster"
+      echo "    bin/install-pr [-k|--kind] ####"
+      installed_cli=2
+      linkerd_check=2
+      exit 0
+      ;;
+    -k|--kind)
+      is_kind=1
+      ;;
+    -?*)
+      echo "Error: Unknown option: $1" >&2
+      exit 1
+      ;;
+    *)
+      break
+  esac
+  shift
+done
+
 pr=$1
 
 if [ -z "$pr" ]
 then
-  echo "usage: ${0##*/} (PR number)" >&2
+  echo "Usage: ${0##*/} (PR number)" >&2
   exit 1
 fi
 
 if [ -z "$GITHUB_TOKEN" ]
 then
   # shellcheck disable=SC2016
-  echo 'Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
+  echo 'Error: Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
   exit 1
 fi
 
+# Get the URL for downloading the artifacts archive.
 auth="Authorization: token $GITHUB_TOKEN"
+branch=$(curl -sL "$linkerd2_pulls_url/$pr" | jq -r '.head.ref')
+artifacts=$(curl -sL "$linkerd2_kind_integration_url/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+archive=$(curl -sL -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
 
-branch=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/pulls/$pr" | jq -r '.head.ref')
-
-artifacts=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
-
-archive_url=$(curl -s -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
-
-dir=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
+bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
+rootdir=$( cd "$bindir"/.. && pwd )
+archive_tmp=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
 
-cd "$dir" || exit
-
-echo "### Downloading images ###"
-
-curl -L -o archive.zip -H "$auth" "$archive_url"
+# Change directory temporarily and download artifact.
+cd "$archive_tmp"
 
+curl -L -o archive.zip -H "$auth" "$archive"
 unzip -o archive.zip
 
-echo "### Pushing images ###"
-
-for archive in cli-bin cni-plugin controller debug grafana proxy web
-do 
-  image=$(docker load -i "$archive.tar" | sed "s/Loaded image: //")
-  docker push "$image"
-  tag=$(echo "$image" | cut -f 2 -d ":")
-done
+# Install the Linkerd CLI.
+image=$(docker load -i cli-bin.tar | sed 's/Loaded image: //')
+container_id=$(docker create "$image")
```

Comment:
We already have `bin/docker-pull-binaries`, why re-create the wheel here?

Reply:
`bin/docker-pull-binaries` doesn't point to the same location as `bin/linkerd`. Maybe those scripts can be merged so that there is only one location where binaries are installed, but I didn't think changing that should be part of this PR.

Ultimately, after this script runs and installs Linkerd, I would expect to use `bin/linkerd` and have it be the version that was installed.

Reply:
we have philosophical differences then :) I expect `bin/linkerd` to build or run the code in my working tree.

Reply:
Ah yep! When using this script I would expect `bin/linkerd` to point to a binary that I'm interested in testing, so I was thinking that this script should therefore put the binary there.

I don't really use the `release/` path in my workflow that much.. pretty much `bin/linkerd` or `linkerd` (the currently installed edge or stable).

I can change that back if that was the intention of two different binary locations, but it does feel a little awkward? If I install a binary that I want to test, it feels odd to have to add or type out the release path location instead of using `bin/linkerd`.

Reply:
I'd prefer to have `bin/linkerd` keep its current behavior of building or running the current working tree.

Reply:
Sounds good!

There will still be a little bit of a dance for choosing the binary since right now this always uses the Darwin binary with `head -n 1`. This script doesn't currently work on Linux. I will keep the `case $(uname)` but go off the `release/` path.